src/Support/Reports/Volumes/ImageAnnotations/CocoReportGenerator.php

```diff
@@ -0,0 +1,164 @@
+                $toZip[$csv->getPath()] = $this->sanitizeFilename("{$id}-{$name}", 'json');
+            }
+        } elseif ($this->shouldSeparateUsers() && $rows->isNotEmpty()) {
+            $rows = $rows->groupBy('user_id');
+            $users = User::whereIn('id', $rows->keys())
+                ->selectRaw("id, concat(firstname, ' ', lastname) as name")
+                ->pluck('name', 'id');
+
+            foreach ($users as $id => $name) {
+                $csv = $this->createCsv($rows->get($id));
+                $this->tmpFiles[] = $csv;
+                $toZip[$csv->getPath()] = $this->sanitizeFilename("{$id}-{$name}", 'json');
+            }
+        } else {
+            $csv = $this->createCsv($rows);
+            $this->tmpFiles[] = $csv;
+            $toZip[$csv->getPath()] = $this->sanitizeFilename("{$this->source->id}-{$this->source->name}", 'json');
+        }
+        $this->executeScript('toCoco', $path);
+        $this->makeZip($toZip, $path);     
+    }
+
+/**
+     * Assemble a new DB query for the volume of this report.
+     *
+     * @return \Illuminate\Database\Query\Builder
+     */
+    protected function query()
+    {
+        $query = $this->initQuery([
+                'image_annotation_labels.id as annotation_label_id',
+                'image_annotation_labels.label_id',
+                'labels.name as label_name',
+                'users.id as user_id',
+                'users.firstname',
+                'users.lastname',
+                'images.id as image_id',
+                'images.filename',
+                'images.lng as longitude',
+                'images.lat as latitude',
+                'shapes.id as shape_id',
+                'shapes.name as shape_name',
+                'image_annotations.points',
+                'images.attrs',
+                'image_annotations.id as annotation_id',
+                'image_annotation_labels.created_at',
+            ])
+            ->join('shapes', 'image_annotations.shape_id', '=', 'shapes.id')
+            ->join('users', 'image_annotation_labels.user_id', '=', 'users.id')
+            ->orderBy('image_annotation_labels.id');
+
+        return $query;
+    }
+
+    /**
+     * Create a CSV file for this report.
+     *
+     * @param \Illuminate\Support\Collection $rows The rows for the CSV
+     * @return CsvFile
+     */
+    protected function createCsv($rows)
+    {
+        $csv = CsvFile::makeTmp();
+        // column headers
+        $csv->put([
+            'annotation_label_id',
+            'label_id',
+            'label_name',
+            'label_hierarchy',
+            'user_id',
+            'firstname',
+            'lastname',
+            'image_id',
+            'filename',
+            'image_longitude',
+            'image_latitude',
+            'shape_id',
+            'shape_name',
+            'points',
+            'attributes',
+            'annotation_id',
+            'created_at',
```

Comment:
Are all of these columns required to generate the COCO file? I don't see `label_hierarchy`, `user_id`, `firstname`, `lastname`, `shape_id`, `annotation_id` and `created_at` in the `toCoco` Python script. If these are not used, they should be removed from the database query and from the temporary CSV file(s).